nxontology_ml/features.py

```diff
@@ -50,3 +50,35 @@ def __init__(self, nxo: NXOntology[str] | None = None):
     def transform(self, X: list[str], copy: bool | None = None) -> NodeFeatures:
         # X are nodes
         return NodeFeatures.from_nodes(nodes=[self._nxo.node_info(n) for n in X])
+
+
+class SubsetsFeatures(DataFrameFnTransformer):
+    def __init__(self, enabled: bool = True) -> None:
+        super().__init__(
+            num_features_fn=self._subset_features,
+            num_features_names=self._feature_names(),
+            enabled=enabled,
+        )
+
+    FILTERED_SUBSETS = [
+        "mondo#disease_grouping",
+        "mondo#gard_rare",
+        "mondo#ordo_etiological_subtype",
+        "mondo#ordo_group_of_disorders",
+    ]
+    FILTERED_SUBSETS_IDX = {s: i for i, s in enumerate(FILTERED_SUBSETS)}
+
+    @classmethod
+    def _subset_features(cls, n: NodeInfo[str]) -> np.array:
+        features = np.zeros(len(cls.FILTERED_SUBSETS), dtype=np.uint8)
+        subsets: list[str] | None = n.data.get("subsets", None)
+        if subsets and len(subsets) > 0:
```

Comment:
Minor:

```suggestion
        subsets: list[str] | None = n.data.get("subsets", [])
        if subsets:
```

Probably don't even need `if subsets`, although I see how its an optimization.

Reply:
Actually `if subsets` is needed: The data has `None` values for most nodes (as opposed to `[]`) and `len(None)` throws the following error:
```python
TypeError: object of type 'NoneType' has no len()
```

Reply:
Ah. A workaround for that is `n.data.get("subsets") or []`, but not necessary (: